catalog/spatial/registry/registry-identification-plugin/src/main/java/org/codice/ddf/registry/identification/RegistryIdPostIngestPlugin.java

```diff
@@ -0,0 +1,161 @@
+
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.registry.common.RegistryConstants;
+import org.codice.ddf.registry.common.metacard.RegistryUtility;
+import org.codice.ddf.security.common.Security;
+import org.opengis.filter.Filter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.operation.CreateResponse;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.UpdateResponse;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PostIngestPlugin;
+
+/**
+ * The RegistryIdPostIngestPlugin collects a set of registry-ids that are currently in the system
+ * and makes them available to the IdentificationPlugin for duplication checking.
+ *
+ * Initially the catalog is queried to initialize the set of registry-ids. After that the list is
+ * maintained by the process methods for create and delete responses.
+ */
+public class RegistryIdPostIngestPlugin implements PostIngestPlugin {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RegistryIdPostIngestPlugin.class);
+
+    private static final int RETRY_INTERVAL = 30;
+
+    private static final int SHUTDOWN_TIMEOUT_SECONDS = 60;
+
+    private static final int PAGE_SIZE = 1000;
+
+    private ScheduledExecutorService executorService;
+
+    private CatalogFramework catalogFramework;
+
+    private FilterBuilder filterBuilder;
+
+    private Set<String> registryIds = ConcurrentHashMap.newKeySet();
+
+    public Set<String> getRegistryIds() {
+        return Collections.unmodifiableSet(registryIds);
+    }
+
+    @Override
+    public CreateResponse process(CreateResponse input) throws PluginExecutionException {
+        if (input != null && input.getCreatedMetacards() != null) {
+            input.getCreatedMetacards()
+                    .stream()
+                    .filter(RegistryUtility::isRegistryMetacard)
+                    .forEach(e -> registryIds.add(RegistryUtility.getRegistryId(e)));
+        }
+        return input;
+    }
+
+    @Override
+    public UpdateResponse process(UpdateResponse input) throws PluginExecutionException {
+        return input;
+    }
+
+    @Override
+    public DeleteResponse process(DeleteResponse input) throws PluginExecutionException {
+        if (input != null && input.getDeletedMetacards() != null) {
+            input.getDeletedMetacards()
+                    .stream()
+                    .filter(RegistryUtility::isRegistryMetacard)
+                    .forEach(e -> registryIds.remove(RegistryUtility.getRegistryId(e)));
+        }
+        return input;
+    }
+
+    public void init() {
```

Comment:
Do we execute catalog queries in the `init()` methods of any other plugins? While this should be okay, it strikes me funny.
1. Start catalog framework
2. Start this plugin, but as part of starting this plugin, execute a query against the catalog
3. Complete starting this plugin so it can be registered to run in the catalog framework